spec.md

```diff
@@ -224,7 +226,12 @@ help intermediate gateways determine how to route the events.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
-  * MUST be unique within the scope of the producer
+  * MUST be unique within the scope of the producer. A producer MUST
+    NOT generate events with identical "source" and "id" values,
+    unless it is generating an identical duplicate of an event that
+    may have been lost due to a transport failure.  A consumer SHOULD
```

Comment:
s/may/might/

Reply:
Does the SHOULD put a big burden on the consumer? How long do they need to hold on to the history of events? I'm wondering if it wouldn't be better to be a "MAY" instead so that we're not forcing people to do anything, rather, just making it clear that spec-wise it's safe to drop dups. Although, there's part of me that wonders if this is even something we should get into at all since we've tended to avoid processing semantics. I think I'd feel more comfortable with:
- producer MUST ensure that source+id is unique for each unique event
- a replayed event (or is it "message", @clemensv ?) MAY include a previously used source+id IFF it is semantically the same
- a consumer MAY interpret an event with a previously seen source+id as a replay of that event

Reply:
+1 the important thing here is that the producer MUST NOT generate events with identical IDs unless it is OK for the consumer to ignore one of them. The consumer can make its own decisions about how it handles duplicates.